python/sglang/srt/managers/schedule_batch.py

```diff
@@ -410,6 +413,72 @@ def jump_forward_and_retokenize(self, jump_forward_str, next_state):
 
         return True
 
+    def jump_forward_and_retokenize_bnf(self, jump_forward_str):
```

Comment:
Build a better abstraction
You can build a new class called `GrammarBackend`. The scheduler just talks to `GrammarBackend` without knowing it is outlines/xgrammar/fsm/bnf..